WordPress/src/main/java/org/wordpress/android/WordPress.java

```diff
@@ -881,55 +820,48 @@ private void onAppComesFromBackground(Activity activity) {
                 mUpdateSelectedSite.runIfNotLimited();
             }
             sDeleteExpiredStats.runIfNotLimited();
-        }
-
-        // gets the note id from the extras that started this activity, so
-        // we can remember to re-set that to unread once the note fetch update takes place
-        private String getNoteIdFromNoteDetailActivityIntent(Intent intent) {
-            String noteId = "";
-            if (intent != null) {
-                noteId = intent.getStringExtra(NotificationsListFragment.NOTE_ID_EXTRA);
-            }
-            return noteId;
-        }
-
-        @Override
-        public void onActivityResumed(Activity activity) {
-            if (sAppIsInTheBackground) {
-                // was in background before
-                onAppComesFromBackground(activity);
-            }
 
-            sAppIsInTheBackground = false;
             if (mFirstActivityResumed) {
-                deferredInit(activity);
+                deferredInit();
             }
             mFirstActivityResumed = false;
         }
+    }
 
+    /**
+     * Uses ComponentCallbacks2 is used for memory-related event handling and configuration changes
+     */
+    private class MemoryAndConfigChangeMonitor implements ComponentCallbacks2 {
         @Override
-        public void onActivityCreated(Activity arg0, Bundle arg1) {
-        }
-
-        @Override
-        public void onActivityDestroyed(Activity arg0) {
-        }
-
-        @Override
-        public void onActivityPaused(Activity arg0) {
-            mLastPingDate = new Date();
+        public void onConfigurationChanged(final Configuration newConfig) {
+            // Reapply locale on configuration change
+            LocaleManager.setLocale(getContext());
         }
 
         @Override
-        public void onActivitySaveInstanceState(Activity arg0, Bundle arg1) {
+        public void onLowMemory() {
         }
 
         @Override
-        public void onActivityStarted(Activity arg0) {
-        }
+        public void onTrimMemory(final int level) {
+            boolean evictBitmaps = false;
+            switch (level) {
+                case TRIM_MEMORY_COMPLETE:
+                case TRIM_MEMORY_MODERATE:
+                case TRIM_MEMORY_RUNNING_MODERATE:
+                case TRIM_MEMORY_RUNNING_CRITICAL:
+                case TRIM_MEMORY_RUNNING_LOW:
+                    evictBitmaps = true;
+                    break;
+                case TRIM_MEMORY_BACKGROUND:
+                case TRIM_MEMORY_UI_HIDDEN:
```

Comment:
Do we need to call `onAppGoesToBackground()` for `TRIM_MEMORY_UI_HIDDEN`?  It was in the previous implementation for that case [here](https://github.com/wordpress-mobile/WordPress-Android/pull/8912/files#diff-17dbba974e19210536213a313661d5a7L768).

Reply:
Great question! Actually the whole point of this PR is to remove that line 😅(I realize now that I didn't explain it that way in the description, so thank you very much for asking this) - I was not really a fan of that idea, and with this PR we're moving away from that model as it has proven to be not as reliable :) - with this PR there's only one point for which each `onAppGoesToBackground()` and `onAppComesFromBackground()` are called and there should be no false positives as there could be with `TRIM_MEMORY_UI_HIDDEN` (there's a nice article about it here https://proandroiddev.com/detecting-when-an-android-app-backgrounds-in-2018-4b5a94977d5c).
We're still relying on `OnTrimMemory` from `ComponentCallbacks2` for memory-related cleanup actions, as that's the actual purpose of it.